x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -269,6 +284,72 @@ export class AlertsClient {
     }
   }
 
+  public async getAlertStatus({ id, dateStart }: GetAlertStatusParams): Promise<AlertStatus> {
+    const eventLogClient = await this.getEventLogClient();
+
+    this.logger.debug('getAlertStatus(): getting the alert');
+    const alert = await this.get({ id });
+    await this.authorization.ensureAuthorized(
+      alert.alertTypeId,
+      alert.consumer,
+      ReadOperations.GetAlertStatus
+    );
+
+    // default duration of status is 60 * alert interval
+    const dateNow = new Date();
+    const durationMillis = parseDuration(alert.schedule.interval) * 60;
+    const defaultDateStart = new Date(dateNow.valueOf() - durationMillis);
+    const parsedDateStart = parseDate(dateStart, 'dateStart', defaultDateStart);
+
+    this.logger.debug('getAlertStatus(): search the event log');
+    let events: IEvent[];
+    try {
+      const queryResults = await eventLogClient.findEventsBySavedObject('alert', id, {
+        page: 1,
+        per_page: 10000,
```

Comment:
This is a bit of a funky API, since you can only provide a start date and not an end date, since the end date is always **NOW** because the data returned also takes into current data (eg, muted instance ids).

I'm not opposed to creating a tech debt issue, especially since it's possible that we could "miss" some data.  I guess the idea would be that we should page through the events till we get to the start date.  I just opened issue https://github.com/elastic/kibana/issues/74860 for this.